ExoCTK/contam_visibility/quaternionx.py

```diff
@@ -1,292 +1,328 @@
-   return Quaternion(V, 0.)
+    """Creates a pure Q, i.e. defines a pointing not a rotation"""
+    return Quaternion(V, 0.)
 
-def cvt_pt_Q_to_V(Q):
-   """Converts a pure (pointing) Q to a unit position Vector"""
-   return Vector(Q.q1, Q.q2, Q.q3)
 
+def cvt_pt_Q_to_V(Q):
+    """Converts a pure (pointing) Q to a unit position Vector"""
+    return Vector(Q.q1, Q.q2, Q.q3)
 
-#The following functions are dependent upon the spacecraft definitions and perhaps should be moved to that module
 
+# The following functions are dependent upon the spacecraft definitions
+# and perhaps should be moved to that module
 def Qmake_body2inertial(coord1, coord2, V3pa):
-   """Creates a rotation Q, going from the body frame to inertial"""
-   return QZ(coord1)*QY(-coord2)*QX(-V3pa)
+    """Creates a rotation Q, going from the body frame to inertial"""
+    return QZ(coord1)*QY(-coord2)*QX(-V3pa)
+
 
 def Qmake_v2v3_2body(v2, v3):
-   """Creates a rotation Q, going from v2 and v3 in the body frame to inertial"""
-   return QY(v3)*QZ(-v2)
+    """Creates a rotation Q, going from v2 and v3 in the body frame to
+    inertial"""
+    return QY(v3)*QZ(-v2)
+
 
 def Qmake_v2v3_2inertial(coord1, coord2, V3pa, v2, v3):
-   """Creates a rotation Q, going from v2 and v3 in the body frame to inertial"""
-   return QZ(coord1)*QY(-coord2)*QX(-V3pa)*QY(v3)*QZ(-v2)
+    """Creates a rotation Q, going from v2 and v3 in the body frame to
+    inertial"""
+    return QZ(coord1)*QY(-coord2)*QX(-V3pa)*QY(v3)*QZ(-v2)
+
+
+def Qmake_aperture2inertial(coord1, coord2, APA, xoff, yoff, s, YapPA,
+                            V3ref, V2ref):
+    """Creates a rotation Q, going from the target in aperture frame to body"""
+    term1 = QZ(coord1)*QY(-coord2)*QX(-APA)*QY(-yoff)
+    term2 = QZ(s*xoff)*QX(YapPA)*QY(V3ref)*QZ(-V2ref)
+    return term1*term2
 
-def Qmake_aperture2inertial(coord1, coord2, APA, xoff, yoff, s, YapPA, V3ref, V2ref):
-   """Creates a rotation Q, going from the target in aperture frame to body"""
-   return QZ(coord1)*QY(-coord2)*QX(-APA)*QY(-yoff)*QZ(s*xoff)*QX(YapPA)*QY(V3ref)*QZ(-V2ref)
 
 def cvt_body2inertial_Q_to_c1c2pa_tuple(Q):
-   """Creates a angle tuple from Q, assuming body frame to inertial Q and 321 rotation sequence"""
-   #Conversion from Euler symmetric parameters to matrix elements and matrix elements to rotation angles
-   # is given in Isaac's papers
-   r11 = Q.q1*Q.q1 - Q.q2*Q.q2 - Q.q3*Q.q3 + Q.q4*Q.q4
-   r21 = 2.*(Q.q1*Q.q2 + Q.q3*Q.q4)
-   r31 = 2.*(Q.q1*Q.q3 - Q.q2*Q.q4)
-   r32 = 2.*(Q.q2*Q.q3 + Q.q1*Q.q4)
-   r33 = -Q.q1*Q.q1 - Q.q2*Q.q2 + Q.q3*Q.q3 + Q.q4*Q.q4
-   coord1 = math.atan2(r21, r11)
-   if coord1 < 0. : coord1 += PI2
-   coord2 = math.asin2(r31)  # use "safe" version of sine
-   pa  = math.atan2(-r32, r33)
-   if pa < 0. : pa += PI2
-   return (coord1, coord2, pa)
+    """Creates a angle tuple from Q, assuming body frame to inertial Q and
+    321 rotation sequence"""

```

Comment:
General comment: 
Move the closing """ in a multiline docstring to its own line (lines 111, 125, and any others)